display/ssd1306.h

```diff
@@ -225,15 +229,39 @@ class SSD1306 : public I2CDevice, Looper, StateMachine, SaberBase, private Audio
 	}
 	if (eof_) {
 	  // STDOUT << "EOF " << frame_count_ << "\n";
-	  screen_ = SCREEN_PLI;
-	  if (frame_count_ == 1) return 5000;
-	  return FillFrameBuffer();
+    if (!SaberBase::IsOn()) {
+	   screen_ = SCREEN_PLI;
+	    if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+	    return FillFrameBuffer();
+    }
 	}
 	frame_count_++;
-	if (looped_frames_ > 1 && millis() - loop_start_ > 5000) {
-	  STDOUT << "END OF LOOP\n";
-	  screen_ = SCREEN_PLI;
-	}
+  if (SaberBase::IsOn()) {
+    // Single frame image
+    if (looped_frames_ == 1) {
+      if (frame_count_ == 1) return font_config.ProffieOSImageDuration;
+      if (frame_count_ > 1) {
+        if (SaberBase::Lockup()) {
+          ShowFile(&IMG_lock);
+        } else {
+          screen_ = SCREEN_PLI;
+          if (!imgon_) {
+            ShowFile(&IMG_on);
+          }
+        }
+      }
+      // looped image
+    } else if (looped_frames_ == frame_count_) {
```

Comment:
While not incorrect, I think it would be better if this was:

} else {
   // looped image
   if (looped_frames_ == frame_count_) {
   }
}

The reason for this is that the else  and the if are testing different concepts (looped vs. non-looped) and (done vs not done)  so joining them together on the same level can make it confusing.